src/script/components/Image.tsx

```diff
@@ -17,83 +17,133 @@
-  asset: AssetRemoteData;
-  assetRepository?: AssetRepository;
-  click?: (asset: AssetRemoteData, event: React.MouseEvent) => void;
+const imageWrapperStyle = {
+  width: '100%',
+};
+
+interface BaseImageProps extends React.HTMLProps<HTMLDivElement> {
+  alt?: string;
   isQuote?: boolean;
   teamState?: TeamState;
 }
 
-const Image: React.FC<ImageProps> = ({
-  asset,
-  click,
+interface RemoteDataImageProps extends BaseImageProps {
+  image: AssetRemoteData;
+  imageSizes?: {width: string; height: string; ratio: number};
+}
+export interface AssetImageProps extends BaseImageProps {
+  image: MediumImage;
+}
+
+export const AssetImage = ({image, alt, ...props}: AssetImageProps) => {
+  const {resource} = useKoSubscribableChildren(image, ['resource']);
+
+  return <Image image={resource} imageSizes={image} alt={alt} {...props} />;
+};
+
+export const Image = ({
+  image,
+  imageSizes,
+  onClick,
   className,
   isQuote = false,
-  assetRepository = container.resolve(AssetRepository),
   teamState = container.resolve(TeamState),
-  aspectRatio,
-  width,
+  alt,
   ...props
-}) => {
+}: RemoteDataImageProps) => {
   const [isInViewport, setIsInViewport] = useState(false);
+  /** keeps track of whether the component is mounted or not to avoid setting the image url in case it's not */
+  const isUnmouted = useRef(false);
 
-  const [assetIsLoading, setAssetIsLoading] = useState<boolean>(false);
-  const [assetSrc, setAssetSrc] = useState<string>();
+  const [imageUrl, setImageUrl] = useState<AssetUrl>();
 
-  const {isFileSharingReceivingEnabled} = useKoSubscribableChildren(teamState, ['isFileSharingReceivingEnabled']);
+  const {getAssetUrl} = useAssetTransfer();
 
-  const onClick = (event: React.MouseEvent) => {
-    if (!assetIsLoading && typeof click === 'function') {
-      click(asset, event);
-    }
-  };
+  const {isFileSharingReceivingEnabled} = useKoSubscribableChildren(teamState, ['isFileSharingReceivingEnabled']);
 
   useEffect(() => {
-    if (isInViewport && isFileSharingReceivingEnabled) {
-      setAssetIsLoading(true);
-      assetRepository.load(asset).then(blob => {
-        if (blob) {
-          setAssetSrc(window.URL.createObjectURL(blob));
+    if (!imageUrl && isInViewport && image && isFileSharingReceivingEnabled) {
+      void (async () => {
+        try {
+          const allowedImageTypes = [
+            'application/octet-stream', // Octet-stream is required to paste images from clipboard
+            ...Config.getConfig().ALLOWED_IMAGE_TYPES,
+          ];
+          const url = await getAssetUrl(image, allowedImageTypes);
+          if (isUnmouted.current) {
+            // Avoid re-rendering a component that is umounted
+            return;
+          }
+          setImageUrl(url);
+        } catch (error) {
+          console.error(error);
         }
-        setAssetIsLoading(false);
-      });
+      })();
     }
+  }, [imageUrl, isInViewport, image, isFileSharingReceivingEnabled, getAssetUrl]);
```

Comment:
This entire loading logic now only exist in a single place: here. 

Previously it was also duplicated in the `ImageAsset` component